pennylane/queuing.py

```diff
@@ -199,28 +199,31 @@ class QueuingManager:
 
     """
 
-    _active_contexts = []
+    _active_context = None
     """The stack of contexts that are currently active."""
 
     @classmethod
     def add_active_queue(cls, queue):
         """Makes a queue the currently active recording context."""
-        cls._active_contexts.append(queue)
+        queue._previous_context = cls._active_context
```

Comment:
What happens if there was a previous previous context?

Reply:
it'll still have the attribute on there! I'm basically turning this python list of AnnotatedQueues into a linked-list

Reply:
Today I learned you can add attributes to instantiated objects in Python :)